mars_troughs/accumulation_model.py

```diff
@@ -16,56 +16,171 @@ class AccumulationModel(Model):
+            meters.
+        """          
+        yt = self.get_yt(time)
+        
+        return -cot_angle * yt + csc_angle * (
+               int_retreat_model_t_spline(time) - int_retreat_model_t_spline(0))
+    
 
 class LinearInsolationAccumulation(InsolationAccumulationModel):
     """
     Accumulation is linear in solar insolation.
+    A(ins(t)) = intercept + slope*ins(t).
+    A is in m/year.
 
     Args:
-        times (np.ndarray): times at which the solar insolation is known
-        insolations (np.ndarray): value of the solar insolations
-        intercept (float, optional): default is 0 millimeters
-        slope (float, optional): default is 1e-6 mm per unit
-            of solar insolation square.
+        times (np.ndarray): times at which the solar insolation is known 
+                            (in years)
+        insolation values (np.ndarray): values of solar insolation (in W/m^2)
+        intercept (float, optional): accumulation rate at present time.
+                                     Default is 1e-6 m/year
+        slope (float, optional): default is 1e-6 m/year per unit
+                                 of solar insolation (m^3/(year*W)).
     """
 
     def __init__(
         self,
         times: np.ndarray,
         insolations: np.ndarray,
-        intercept: float = 1.0,
+        intercept: float = 1e-6,
         slope: float = 1e-6,
-    ):
+        ):
         super().__init__(times, insolations)
         self.intercept = intercept
         self.slope = slope
 
     @property
-    def parameters(self) -> Dict[str, float]:
-        return {"intercept": self.intercept, "slope": self.slope}
+    def parameter_names(self) -> Dict[str, float]:
+        return ["intercept", "slope"]
 
-    def accumulation(self, time: np.ndarray) -> np.ndarray:
-        raise NotImplementedError
+    def get_accumulation_at_t(self, time: np.ndarray) -> np.ndarray:
+        """
+        Calculates the accumulation rate at times "time". 
+
+        Args:
+            time (np.ndarray): times at which we want to calculate A, in years.
+        Output:
+            np.ndarray of the same size as time input containing values of 
+            accumulation rates A, in m/year
+        
+        """
+        return self.intercept + self.slope * self._ins_data_spline(time)
+    
+    def get_yt(self, time: np.ndarray):
+        """
+        Calculates the vertical distance y (in m) traveled by a point
+        in the center of the high side of the trough. This distance  is a 
+        function of the accumulation rate A as y(t)=int(A(ins(t)), dt) or 
+        dy/dt=A(ins(t))
+
+        Args:
+            time (np.ndarray): times at which we want to calculate y, in years.
+        Output:
+            np.ndarray of the same size as time input containing values of 
+            the vertical distance y, in meters.
+        
+        """
+        return  -1 *(self.slope * (self._int_ins_data_spline(time) 
```

Comment:
Shouldn't this have the intercept somewhere? I was looking at the same equation for the quadratic model and it has the intercept.

Reply:
Yep, this was wrong. It should be y=-(intercept*time+slope*int_ins_data_spline(time)).